android/src/main/java/com/bitmovin/player/reactnative/BufferModule.kt

```diff
@@ -2,28 +2,28 @@ package com.bitmovin.player.reactnative
 
 import com.bitmovin.player.api.buffer.BufferLevel
 import com.bitmovin.player.api.media.MediaType
-import com.bitmovin.player.reactnative.converter.JsonConverter
+import com.bitmovin.player.reactnative.converter.toBufferType
+import com.bitmovin.player.reactnative.converter.toJson
 import com.facebook.react.bridge.*
 import com.facebook.react.module.annotations.ReactModule
-import com.facebook.react.uimanager.UIManagerModule
 
 private const val MODULE_NAME = "BufferModule"
 
 @ReactModule(name = MODULE_NAME)
-class BufferModule(private val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+class BufferModule(context: ReactApplicationContext) : BitmovinBaseModule(context) {
     override fun getName() = MODULE_NAME
 
     /**
      * Gets the [BufferLevel] from the Player
      * @param nativeId Target player id.
-     * @param type The [type of buffer][JsonConverter.toBufferType] to return the level for.
+     * @param type The [type of buffer][toBufferType] to return the level for.
      * @param promise JS promise object.
      */
     @ReactMethod
     fun getLevel(nativeId: NativeId, type: String, promise: Promise) {
-        uiManager()?.addUIBlock { _ ->
-            val player = playerModule()?.getPlayer(nativeId) ?: return@addUIBlock
-            val bufferType = JsonConverter.toBufferType(type)
+        addUIBlock(promise) {
```

Comment:
I think we might have a small problem with this function and block.
The `addUiBlock` function automatically resolves the `promise` with the return value of the `block`. However, here the `block` is also executing code that resolves or rejects the promise 🤔 
Further, the `return` statement one line below would trigger a `resolve(Unit)` in the `addUiBlock`, which might be not expected. (It's already a lot better than before though, as with the new solution we at least resolve the promise eventually)

I think we need to find a contract (best would be compile time save), to handle those situations. (`player` being `null` should probably also reject the promise)

Reply:
This is partially fixed:
 - null player rejects the promise
 - promise is now typed, so unit can't be returned by mistake

The one thing that I could not fix (and I suspect is not fixable without affine types) is the code calling `promise.resolve` explicitly instead of using the provided helpers.
I made sure no Promise.resolve was left in the code.